example/mutatingwebhook.go

```diff
@@ -46,24 +46,40 @@ func (a *podAnnotator) Handle(ctx context.Context, req types.Request) types.Resp
 	}
 	copy := pod.DeepCopy()
 
-	err = mutatePodsFn(ctx, copy)
+	err = a.mutatePodsFn(ctx, copy)
 	if err != nil {
 		return admission.ErrorResponse(http.StatusInternalServerError, err)
 	}
 	return admission.PatchResponse(pod, copy)
 }
 
 // mutatePodsFn add an annotation to the given pod
-func mutatePodsFn(ctx context.Context, pod *corev1.Pod) error {
-	v, ok := ctx.Value(admission.StringKey("foo")).(string)
-	if !ok {
-		return fmt.Errorf("the value associated with %v is expected to be a string", "foo")
-	}
+func (a *podAnnotator) mutatePodsFn(ctx context.Context, pod *corev1.Pod) error {
 	anno := pod.Annotations
```

Comment:
this can be shortened to just
if pod.Annotations == nil {
  pod.Annotations = map[string]string{}
}
pod.Annotations["xyz"] = "qpr"